pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjMethodRoute.java

```diff
@@ -0,0 +1,133 @@
+    @NonNull private final Counter failedGrpcRequestCounter;
+    @NonNull private final Counter failedHttpRequestCounter;
+    @NonNull private final Counter failedUnknownRequestCounter;
+    @NonNull private final Counter deadlineExceededCounter;
+    
+    PbjMethodRoute(
+            @NonNull final ServiceInterface service,
+            @NonNull final ServiceInterface.Method method) {
+        this.service = requireNonNull(service);
+        this.method = requireNonNull(method);
+
+        final var serviceName = service.fullName();
+        final var methodName = method.name();
+        this.fullPath = serviceName + SEP + methodName;
+        this.pathMatcher = PathMatchers.exact(fullPath);
+
+        final var metricRegistry = Metrics.globalRegistry();
+        this.requestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.requests")
+                        .scope(SCOPE)
+                        .addTag(Tag.create(SERVICE_TAG, serviceName))
+                        .addTag(Tag.create(METHOD_TAG, methodName))
+                        .description("The number of gRPC requests"));
+        this.failedGrpcRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.failed.requests")
+                        .scope(SCOPE)
+                        .addTag(Tag.create(SERVICE_TAG, serviceName))
+                        .addTag(Tag.create(METHOD_TAG, methodName))
+                        .addTag(Tag.create(FAILURE_TAG, "grpc-exception"))
+                        .description("The number of failed gRPC requests"));
+        this.failedHttpRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.failed.requests")
+                        .scope(SCOPE)
+                        .addTag(Tag.create(SERVICE_TAG, serviceName))
+                        .addTag(Tag.create(METHOD_TAG, methodName))
+                        .addTag(Tag.create(FAILURE_TAG, "http-exception"))
+                        .description("The number of failed gRPC requests"));
+        this.failedUnknownRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.failed.requests")
+                        .scope(SCOPE)
+                        .addTag(Tag.create(SERVICE_TAG, serviceName))
+                        .addTag(Tag.create(METHOD_TAG, methodName))
+                        .addTag(Tag.create(FAILURE_TAG, "unknown-exception"))
+                        .description("The number of failed gRPC requests"));
+        this.deadlineExceededCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.deadline.exceeded")
+                        .scope(SCOPE)
+                        .addTag(Tag.create(SERVICE_TAG, serviceName))
+                        .addTag(Tag.create(METHOD_TAG, methodName))
+                        .description("The number of gRPC requests that exceeded their deadline"));
+    }
+
+    @Override
+    @NonNull
+    PbjMethodRoute toPbjMethodRoute(@NonNull final HttpPrologue grpcPrologue) {
+        return this;
```

Comment:
The method is required by the abstract parent class. Since a sibling implementation uses the parameter, it can't be removed.

I don't think there are any checks that need to be done here. The method is `toPbjMethodRoute`, on the `PbjMethodRoute` class. So it can just reflexively return itself.

Reply:
The provided `grpcPrologue` may describe a route that differs from the route of the `this` object, so there's some room for verification.

Thanks for clarifying about the sibling implementation.